app/services/canvas_service.py

```diff
@@ -181,29 +181,32 @@ async def _upload_file(
         self,
         upload_url: str,
         file: BinaryIO,
-        parent_folder_path_or_id: str | int,
-        on_duplicate: DuplicateFileAction,
-        *,
-        headers={}
+        parent_folder_path_or_id: Path | str | int | None = None,
+        on_duplicate: DuplicateFileAction = DuplicateFileAction.OVERWRITE
     ):
         use_id = isinstance(parent_folder_path_or_id, int)
+        use_path = isinstance(parent_folder_path_or_id, (str, Path))
```

Comment:
I don't think isinstance does a None check, so you still need to do that since you've added the ability for `parent_folder_path_or_id` to be None.

Reply:
if you look a bit further down no action is taken if the value is None.

Reply:
Oh, I was thinking about the use of `use_path` on line 201, but yea if you just do `str(None)` you just get `None` so it's fine

Reply:
Maybe I misunderstood what you meant in your first comment. Isinstance definitely checks NoneType. isinstance(None, <anything_other_than_NoneType>) is false. So neihter use_path or use_id will ever be true when the arg is none

Reply:
Oh that's fair too, it does do what I was wanting, yea. Not sure what I thought now 🤷‍♂️